### PR TITLE
Migrate rex to this repository

### PR DESCRIPTION
**Pull request by mheinzel**
See https://github.com/zinfra/backend-issues/issues/1264.

I have not tried running it yet, but it compiles with the current snapshot.

### CODE REVIEW

File: tools/rex/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM alpine:3.5
```
Comment (jschaul) [inline tools/rex/Dockerfile:1]: let's update this

File: tools/rex/Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+default: dist
+
+.PHONY: dist
+dist: clean $(ACI)
+
+.PHONY: build
+build: rex
+
+.PHONY: publish
+publish: $(ACI) $(ACI).asc s3-exists/$(ACI) s3-upload/$(ACI)
```
Comment (jschaul) [inline tools/rex/Makefile:20]: some of those targets don't exist anymore?

File: tools/rex/rex.cabal [hunk 3]
```diff
@@ -0,0 +1,46 @@
+name:                rex
+version:             0.3.0
+synopsis:            Scrape and expose restund metrics for prometheus
+description:         @rex@ = @restund-exporter@ in prometheus jargon
+author:              Kim Altintop
+maintainer:          kim@wire.com
```
Comment (jschaul) [inline tools/rex/rex.cabal:6]: this should be changed to Wire Swiss. Also add a license.

File: tools/rex/.stylish-haskell.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      list_padding: 4
+      separate_lists: true
+  - language_pragmas:
+      style: vertical
+      align: true
+      remove_redundant: true
+  - tabs:
+      spaces: 4
+  - trailing_whitespace: {}
+columns: 80
```
Comment (akshaymankar) [inline tools/rex/.stylish-haskell.yaml:20]: Let's remove this and ormolu the code.

**Review by akshaymankar — CHANGES_REQUESTED** [review]

File: tools/rex/.stylish-haskell.yaml (hunk 4 above)
Comment (mheinzel) [reply tools/rex/.stylish-haskell.yaml:20]: It's ormulized, but yes, I still need to clean up some files.

File: tools/rex/Dockerfile (hunk 1 above)
Comment (mheinzel) [reply tools/rex/Dockerfile]: done

File: tools/rex/rex.cabal (hunk 3 above)
Comment (mheinzel) [reply tools/rex/rex.cabal]: done

File: tools/rex/.stylish-haskell.yaml (hunk 4 above)
Comment (mheinzel) [reply tools/rex/.stylish-haskell.yaml]: removed this and some more stuff.

File: tools/rex/Makefile (hunk 2 above)
Comment (mheinzel) [reply tools/rex/Makefile:20]: Yeah, and a few more things need fixing.
- there's an include that doesn't work anymore
- we should probably stack install into `dist` as with the other tools

I'm not quite sure how rex is supposed to be packaged/used, so maybe someone else could take care of these outstanding issues with the Makefile?

**Comment by mheinzel** [comment]
I fixed all of the issues except for the Makefile. Once that's done, this PR should be ready.

> 2 out of 3 committers have signed the CLA.

Kim worked for Wire when writing this code, so we can go ahead without the CLA.

**Comment by jschaul** [comment]
I added two commits here to put the Makefile in order, add a README, and also to compile a static version based on the current GHC/alpine docker image.

**Review by jschaul — APPROVED** [review]

**Comment by jschaul** [comment]
@akshaymankar any further concerns?

**Review by akshaymankar — APPROVED** [review]
Looks good now.

**Pull request merged**